homeassistant/components/sensor/efergy.py

```diff
@@ -102,6 +106,12 @@ def update(self):
                 url_string = _RESOURCE + 'getInstant?token=' + self.app_token
                 response = get(url_string)
                 self._state = response.json()['reading'] / 1000
+            elif self.type == 'amount':
+                url_string = _RESOURCE + 'getEnergy?token=' + self.app_token \
+                    + '&offset=' + self.utc_offset + '&period=' \
+                    + self.period
+                response = get(url_string)
```

Comment:
Can you please add a timeout to the request? Would be appreciated if you do this for the other requests as well.